src/modules/dashboard/index.tsx

```diff
@@ -83,55 +80,95 @@ const Dashboard = () => {
     setRowsPerPage(parseInt(event.target.value, 10));
     setPage(0);
   };
+  const { data: categoriesList, isLoading: listFetching } =
+  useCategories(departmentId);
+  const categoryOptions = useMemo(() => {
+    return (
+      categoriesList?.map((cate) => ({
+        label: cate.name,
+        value: cate.name,
+      })) || []
+    );
+  }, [categoriesList]);
+const { data: departmentsList, isLoading: departmentsFetching } =
+  useDepartments(1);
+
+const deptOptions = useMemo(() => {
+  return (
+    departmentsList?.map((dept) => ({
+      label: dept.name,
+      value: dept.id,
+    })) || []
+  );
+}, [departmentId, departmentsList]);
 
   const updatedData = data?.slice(page * rowsPerPage, (page + 1) * rowsPerPage);
+  
 
   return (
-    <div>
-      <div className='d-flex flex-column p-5 '>
-        <div className='d-flex gap-3 mb-4'>
-          <CustomSelect
+    <Box sx={{display: 'flex', flexDirection: 'column'}}>
+      <Box sx={{display: 'flex', gap: '1.5rem', mb: '1.5rem'}} className='complaint-card-filters'>
+        <Box sx={{display: 'grid', gap: '1.5rem'}} className='filter-input-group flex-1'>
+         {userAuth.role !== "employee" && <CustomSelect
             label={'Status'}
             options={statusOptions}
             value={filters.status}
             onChange={handleChange}
             name='status'
-          />
-          <CustomSelect
-            label={'Departments'}
-            options={departmentOptions}
+          />}
+           {userAuth.role !== "employee" &&  <CustomSelect
+            label={'departments'}
+            options={deptOptions}
             value={filters.department}
+            onChange={(e)=>{setDepartmentId(+(e.target.value));
```

Comment:
create separate function here & avoid inline functions

Reply:
here we want to convert into string -> integer (only for this case )